### ISSUE TITLE
[2.3.3] RxReturnValueIgnored false-positive

### ISSUE THREAD

**Issue by boris-petrov** [google/error-prone#1230]
```java
ConcurrentMap<String, Single<String>> map = new ConcurrentHashMap<>();
map.remove("foobar");
```

The second line will give a `RxReturnValueIgnored` warning (because `remove` returns a `Single<String>`) which is wrong.

**Comment by graememorgan** [comment]
Wow, that's a curious case. Presumably the same would, in principle, apply to any other ReturnValueIgnored checks.

Have you seen this in practice?

**Comment by boris-petrov** [comment]
ReturnValueIgnored is actually one of the very few warnings I’ve disabled, because I get lots and lots of false-positives. I could point out some next week if you would like.

**Comment by ZacSweers** [comment]
Maybe remove/put should be implicit `@CanIgnoreReturnValue` types?
